### PR TITLE
Apply new symbolic shape strategy to make_fx symbolic mode

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85260

This results in some test wobbling, which looks legit.  I also
added some debug helpers for stuff that I found useful while
working on this.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -806,7 +806,8 @@ def f(x):
         shape_env = self._test_dynamic(f, [(3, 4)], test_inputs)
         self.assertTrue(shape_env.evaluate_guards_for_args(torch.randn(4, 5)))
         self.assertFalse(shape_env.evaluate_guards_for_args(torch.randn(25, 5)))
-        assert len(shape_env.guards) == 1
+        # one guard for size/stride contiguity, and one substantive guard
+        assert len(shape_env.guards) == 2, "\n" + shape_env.format_guards()
```
Comment (albanD) [inline test/test_proxy_tensor.py:810]: self.assertEqual

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -894,8 +895,8 @@ def f(a, b):
         self.assertTrue(meta_c.meta['val'].shape[0].get_pyobj() == meta_d.meta['val'].expr)
 
     def _assert_no_guards(self, fx_g, free_symbols):
-        self.assertEqual(_get_free_symbols(fx_g.shape_env), free_symbols)
-        self.assertEqual(len(fx_g.shape_env.get_nontrivial_guards()), 0)
+        assert _get_free_symbols(fx_g.shape_env) == free_symbols, fx_g.shape_env.var_to_val
```
Comment (albanD) [inline test/test_proxy_tensor.py:898]: Why this change? We usually ask that no plain assert is ever used in the test suite.

**Review by albanD — APPROVED** [review]
Ok!

File: test/test_proxy_tensor.py (hunk 1 above)
Comment (ezyang) [reply test/test_proxy_tensor.py:810]: Plain assert is intentional; fir some reason assertEqual is suppressing msg
